contracts/src/v0.8/shared/test/testhelpers/GasConsumer.sol

```diff
@@ -0,0 +1,25 @@
+pragma solidity ^0.8.0;
+
+contract GasConsumer {
+  function consumeAllGas() external view {
+    assembly {
+      for {
```

Comment:
given it is YUL, please add a comment to describe what loop does, and any magic numbers in it.
E.g. 
why terminate at gas() < 30
why do the `pop(add(0, 0))`